### PR TITLE
Sygma iteration 2 - Made the BuildApprovals method. Added method in contract that helps me build a transaction request

### PR DESCRIPTION
**Pull request by kantagara**

### CODE REVIEW

File: src/ChainSafe.Gaming/RPC/Contracts/Contract.cs [hunk 1]
```diff
@@ -161,20 +161,8 @@ public async Task<object[]> Send(string method, object[] parameters = null, Tran
             parameters ??= Array.Empty<object>();
 
             var function = contractBuilder.GetFunctionBuilder(method);
-            var txReq = overwrite ?? new TransactionRequest();
```
Comment (kalambet) [inline src/ChainSafe.Gaming/RPC/Contracts/Contract.cs:164]: I think this change breaks the original logic. The optional assignment for `From`, `To`, `Data` and etc was there in the case provided `overwrite` object does not have any.

So it was possible to provide some user-defined fields in this method. But with this logic `overwrite` object is not used at all.

File: src/ChainSafe.Gaming/RPC/Contracts/Contract.cs [hunk 2]
```diff
@@ -88,7 +88,7 @@ public async Task<object[]> Call(string method, object[] parameters = null, Tran
             var function = contractBuilder.GetFunctionBuilder(method);
             var txReq = overwrite ?? new TransactionRequest();
             txReq.To ??= address;
-            txReq.From ??= await signer.GetAddress();
```
Comment (kalambet) [inline src/ChainSafe.Gaming/RPC/Contracts/Contract.cs:91]: Maybe adopt `PrepareTransactionRequest` here as well?

File: src/ChainSafe.Gaming/RPC/Contracts/Contract.cs (hunk 1 above)
Comment (kantagara) [reply src/ChainSafe.Gaming/RPC/Contracts/Contract.cs:164]: You're right! Fixed

**Review by rob1997 — APPROVED** [review]
some DRY in `PrepareTransactionRequest` method, LGTM!

**Review by sneakzttv — APPROVED** [review]

**Review by kalambet — APPROVED** [review]
Nice!

**Review by kalambet — APPROVED** [review]

**Pull request merged**